lib/models/addons-factory.js

```diff
@@ -35,9 +35,15 @@ class AddonsFactory {
     logger.info('     addon names are:', Object.keys(addonPackages));
 
     for (let name in addonPackages) {
+      console.log(`### factory.initAddons for '${name}'`);
       addonInfo = addonPackages[name];
       emberAddonConfig = addonInfo.pkg['ember-addon'];
 
+      if (!emberAddonConfig) {
```

Comment:
this seem suspicious, how would other addons positions themselves before/after an existing addon, if its ember-addon field was empty/missing?

Reply:
Oops - I forgot to remove that. Good catch! 

At one point I'd missed a step in the previous version of addon-discovery where it sets the 'ember-addon' field to {} if it wasn't present. This resulted in an NPE here. I added the check to see what was going on. Then I fixed PackageInfo to do the relevant setting during the ctor so the check isn't needed anymore. I've removed it and will check in the fix.